src/elements/table_handler.js

```diff
@@ -0,0 +1,480 @@
+      if (!firstCell) return
+
+      const currentStyle = firstCell.getHeaderStyles()
+      const newStyle = currentStyle ^ TableCellHeaderStates.ROW
+
+      cells.forEach(cell => {
+        const tableCellNode = $getTableCellNodeFromLexicalNode(cell)
+
+        if (tableCellNode) {
+          tableCellNode.setHeaderStyles(newStyle, TableCellHeaderStates.ROW)
+        }
+      })
+    })
+  }
+
+  #toggleColumnHeaderStyle() {
+    this.#editor.update(() => {
+      const rows = this.currentTableNode.getChildren()
+
+      const row = rows[this.#currentRow]
+      if (!row) return
+
+      const cells = row.getChildren()
+      const selectedCell = $getTableCellNodeFromLexicalNode(cells[this.#currentColumn])
+      if (!selectedCell) return
+
+      const currentStyle = selectedCell.getHeaderStyles()
+      const newStyle = currentStyle ^ TableCellHeaderStates.COLUMN
+
+      rows.forEach(row => {
+        const cell = row.getChildren()[this.#currentColumn]
+        if (!cell) return
+
+        const tableCellNode = $getTableCellNodeFromLexicalNode(cell)
+
+        if (tableCellNode) {
+          tableCellNode.setHeaderStyles(newStyle, TableCellHeaderStates.COLUMN)
+        }
+      })
+    })
+  }
+
+  #monitorForTableSelection() {
+    this.#editor.registerUpdateListener(() => {
+      this.#editor.getEditorState().read(() => {
+        const tableNode = this.#getCurrentTableNode()
+
+        if (tableNode) {
+          this.#tableCellWasSelected(tableNode)
+        } else {
+          this.#hideButtons()
+        }
+      })
+    })
+  }
+
+  #getCurrentTableNode() {
+    const selection = $getSelection()
+
+    if (!$isRangeSelection(selection)) {
+      return null
+    }
+
+    const anchorNode = selection.anchor.getNode()
+    return $findTableNode(anchorNode)
+  }
+
+  #selectLastTableCell() {
+    const tableNode = this.#getCurrentTableNode()
+    if (!tableNode) return
+
+    const last = tableNode.getLastChild().getLastChild()
+    if (!$isTableCellNode(last)) return
+
+    last.selectEnd()
+  }
+
+  #tableCellWasSelected(tableNode) {
+    this.currentTableNode = tableNode
+    this.#updateButtonsPosition(tableNode)
+    this.#showButtons()
+  }
+
+  #updateButtonsPosition(tableNode) {
+    const tableElement = this.#editor.getElementByKey(tableNode.getKey())
+    if (!tableElement) return
+
+    const tableRect = tableElement.getBoundingClientRect()
+    const editorRect = this.#editorElement.getBoundingClientRect()
+
+    const relativeTop = tableRect.top - editorRect.top
+    const relativeCenter = (tableRect.left + tableRect.right) / 2 - editorRect.left
+    this.buttonsContainer.style.top = `${relativeTop}px`
+    this.buttonsContainer.style.left = `${relativeCenter}px`
+  }
+
+  #setTableFocusState(focused) {
+    this.#editorElement.querySelector("div.node--selected:has(table)")?.classList.remove("node--selected")
+
+    if (focused) {
+      if (!this.currentTableNode) return
```

Comment:
could do `if (focused && this.currentTablenode)` and remove the guard return completely.